src/main/java/ScrabbleFrame.java

```diff
@@ -0,0 +1,94 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+
+public class ScrabbleFrame extends JFrame{
+    private int score = 0;
+    private final JLabel scoreLabel;
+    private JLabel[] tiles;
+    private final JTextField inputField;
+    private final JButton submitButton;
+    private final JLabel outputLabel;
+    private JPanel verticalPanel;
+    private ScrabbleGame scrabbleGame;
+    private ScrabbleDictionary dictionary;
+
+
+    public ScrabbleFrame()
+    {
+        setTitle("Touro Scrabble");
+        setSize(800,600);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+        setLayout(new FlowLayout());
+
+        dictionary = new ScrabbleDictionary();
+        LetterPool letterPool = new LetterPool();
+        scrabbleGame = new ScrabbleGame(dictionary, letterPool);
+
+        verticalPanel = new JPanel();
+        verticalPanel.setLayout(new BoxLayout(verticalPanel,BoxLayout.Y_AXIS));
+        add(verticalPanel);
+
+        addTilesPanel(verticalPanel);
+
+        inputField = new JTextField();
+        inputField.setPreferredSize(new Dimension(120,60));
+        verticalPanel.add(inputField);
+
+        scoreLabel = new JLabel("Score: "+ score);
+        scoreLabel.setPreferredSize(new Dimension(120, 60));
+        verticalPanel.add(scoreLabel);
+
+        submitButton = new JButton("Submit");
+        submitButton.setPreferredSize(new Dimension(120,60));
+        submitButton.addActionListener(this::onSubmitClicked);
+        verticalPanel.add(submitButton);
+
+        outputLabel = new JLabel("");
+        outputLabel.setPreferredSize(new Dimension(120, 60));
+        verticalPanel.add(outputLabel);
+
+    }
+
+    private void addTilesPanel(JPanel verticalPanel) {
+        JPanel tilesPanel = new JPanel();
+        tilesPanel.setLayout(new FlowLayout());
+        tiles = new JLabel[7];
+        for (int i = 0; i < 7; i++) {
+            tiles[i] = new JLabel(scrabbleGame.tiles.get(i).toString());
+            tilesPanel.add(tiles[i]);
+        }
+        verticalPanel.add(tilesPanel);
+    }
+
+    public void onSubmitClicked(ActionEvent actionEvent)
+    {
+        outputLabel.setText("");
+        String word = inputField.getText();
+        if (!dictionary.isWord(word))
+        {
+            outputLabel.setText("Not a valid word");
+        }
+        else if (scrabbleGame.playWord(word))
```

Comment:
You are looking up the weird twice. Once in here and once in the code above.

Reply:
Can I set the playWord method return a string that I can display here instead of having it figure out why it failed in the ScrabbleFrame class?

Reply:
Does that make sense?

Reply:
I just wanted to check if you were okay with us changing the ScrabbleGame class. Since I'd also need to change all of my tests, I figured I should make sure it's a good idea before I try it.